packages/wonder-blocks-dropdown/components/dropdown.test.js

```diff
@@ -0,0 +1,379 @@
+    });
+
+    it("focuses correct item with a disabled item", () => {
+        dropdown.setProps({
+            initialFocusedIndex: 0,
+            items: [
+                {
+                    component: <OptionItem label="item 0" value="0" key="0" />,
+                    focusable: true,
+                    populatedProps: {},
+                },
+                {
+                    component: <OptionItem label="item 1" value="1" key="1" />,
+                    focusable: false,
+                    populatedProps: {},
+                },
+                {
+                    component: <OptionItem label="item 2" value="2" key="2" />,
+                    focusable: true,
+                    populatedProps: {},
+                },
+            ],
+            keyboard: true,
+            open: true,
+        });
+
+        const option0 = dropdown.find("OptionItem").at(0);
+        const option2 = dropdown.find("OptionItem").at(2);
+
+        expect(dropdown.instance().focusedIndex).toBe(0);
+
+        // Should select option2
+        option0.simulate("keydown", {keyCode: keyCodes.down});
+        option0.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(1);
+        expect(dropdown.instance().focusedOriginalIndex).toBe(2);
+
+        // Should select option0
+        option2.simulate("keydown", {keyCode: keyCodes.down});
+        option2.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(0);
+        expect(dropdown.instance().focusedOriginalIndex).toBe(0);
+    });
+
+    it("focuses correct item after different items become focusable", () => {
+        dropdown.setProps({
+            initialFocusedIndex: 0,
+            keyboard: true,
+            open: true,
+        });
+
+        const option0 = dropdown.find("OptionItem").at(0);
+
+        expect(dropdown.instance().focusedIndex).toBe(0);
+
+        // Should select option1
+        option0.simulate("keydown", {keyCode: keyCodes.down});
+        option0.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(1);
+
+        dropdown.setProps({
+            items: [
+                {
+                    component: <OptionItem label="item 0" value="0" key="0" />,
+                    focusable: false,
+                    populatedProps: {},
+                },
+                {
+                    component: <OptionItem label="item 1" value="1" key="1" />,
+                    focusable: true,
+                    populatedProps: {},
+                },
+                {
+                    component: <OptionItem label="item 2" value="2" key="2" />,
+                    focusable: true,
+                    populatedProps: {},
+                },
+            ],
+        });
+
+        // Should figure out that option1, which is now at index 0, is selected
+        expect(dropdown.instance().focusedIndex).toBe(0);
```

Comment:
It keeps track of the select item while updating the items in the list?!  What happens if the item that was selected is removed from the listed while focused?

Reply:
Good call. What's supposed to happen is that the initial first item gains focus. I'll add a test.

Reply:
Oh, that's the next test, but I'll make it more explicit.